pallet/pallet-codegen/src/main/java/com.strategyobject.substrateclient.pallet/StorageProcessor.java

```diff
@@ -0,0 +1,440 @@
+    private static final String HASHERS = "hashers";
+    private static final String INTERNAL_SUFFIX = "Internal";
+
+    public StorageProcessor(@NonNull TypeElement palletElement) {
+        super(palletElement);
+    }
+
+    @Override
+    void process(@NonNull String palletName,
+                 @NonNull ExecutableElement method,
+                 TypeSpec.@NonNull Builder typeSpecBuilder,
+                 MethodSpec.Builder constructorBuilder,
+                 @NonNull ProcessorContext context) throws ProcessingException {
+        val annotation = AnnotationUtils.getAnnotationMirror(method, Storage.class);
+        if (annotation == null) {
+            return;
+        }
+
+        validate(method, annotation, context);
+
+        createBackField(typeSpecBuilder, method);
+        typeSpecBuilder.addMethod(privateMethod(palletName, method, annotation, context));
+        assignBackFieldInConstructor(constructorBuilder, method);
+        typeSpecBuilder.addMethod(publicMethod(method));
+    }
+
+    private void createBackField(TypeSpec.Builder typeSpecBuilder, ExecutableElement method) {
+        typeSpecBuilder.addField(
+                TypeName.get(method.getReturnType()),
+                method.getSimpleName().toString(),
+                Modifier.FINAL, Modifier.PRIVATE
+        );
+    }
+
+    private void assignBackFieldInConstructor(MethodSpec.Builder constructorBuilder, ExecutableElement method) {
+        constructorBuilder.addStatement("this.$1L = $1L$2L()",
+                method.getSimpleName().toString(),
+                INTERNAL_SUFFIX);
+    }
+
+    private MethodSpec publicMethod(ExecutableElement method) {
+        val returnType = method.getReturnType();
+
+        return MethodSpec.methodBuilder(method.getSimpleName().toString())
+                .addModifiers(Modifier.PUBLIC)
+                .addAnnotation(Override.class)
+                .returns(TypeName.get(returnType))
+                .addStatement("return this.$L", method.getSimpleName().toString())
+                .build();
+    }
+
+    private MethodSpec privateMethod(String palletName,
+                                     ExecutableElement method,
+                                     AnnotationMirror storageAnnotation,
+                                     ProcessorContext context) throws ProcessingException {
+        val returnType = method.getReturnType();
+        validateMethodSignature(method, returnType, context);
+
+        val valueType = ((DeclaredType) returnType).getTypeArguments().get(0);
+
+        val methodSpecBuilder =
+                MethodSpec.methodBuilder(method.getSimpleName().toString() + INTERNAL_SUFFIX)
+                        .addModifiers(Modifier.PRIVATE)
+                        .addAnnotation(suppressWarnings("unchecked", "rawtypes"))
+                        .returns(TypeName.get(returnType));
+
+        declareReaderAndWriterRegistries(methodSpecBuilder);
```

Comment:
cannot it be moved upper to a class level? registries are threadsafe and needed by multiple implementations.

Reply:
I think it's not necessary because there are just variables which refer to singletons.